packages/edit-post/src/store/actions.js

```diff
@@ -240,24 +259,128 @@ export function showBlockTypes( blockNames ) {
  *
  * @param {Object} metaBoxesPerLocation Meta boxes per location.
  *
- * @return {Object} Action object.
+ * @yield {Object} Action object.
  */
-export function setAvailableMetaBoxesPerLocation( metaBoxesPerLocation ) {
-	return {
+export function* setAvailableMetaBoxesPerLocation( metaBoxesPerLocation ) {
+	yield {
 		type: 'SET_META_BOXES_PER_LOCATIONS',
 		metaBoxesPerLocation,
 	};
+
+	const postType = yield controls.select(
+		'core/editor',
+		'getCurrentPostType'
+	);
+	if ( window.postboxes.page !== postType ) {
+		window.postboxes.add_postbox_toggles( postType );
+	}
+
+	let wasSavingPost = yield controls.select( 'core/editor', 'isSavingPost' );
+	let wasAutosavingPost = yield controls.select(
+		'core/editor',
+		'isAutosavingPost'
+	);
+
+	// Meta boxes are initialized once at page load. It is not necessary to
+	// account for updates on each state change.
+	//
+	// See: https://github.com/WordPress/WordPress/blob/5.1.1/wp-admin/includes/post.php#L2307-L2309
+	const hasActiveMetaBoxes = yield controls.select(
+		'core/edit-post',
+		'hasMetaBoxes'
+	);
+
+	// First remove any existing subscription in order to prevent multiple saves
+	if ( !! saveMetaboxUnsubscribe ) {
```

Comment:
For this to work you'd need to declare `let saveMetaboxUnsubscribe` outside of this function, right now this will always be false since this variable is undefined at this point. I think this doesn't throw an error only because `const` is transpiled to `var`.

Reply:
BTW this won't work with multiple registries. However, effects.js wouldn't as well so I don't think it's a blocker - just something to take not of.

Reply:
I updated that in 2e12440

Reply:
Thanks, @adamziel, that makes sense 👍